smoke_tests/smoke_tests.py

```diff
@@ -0,0 +1,92 @@
+import argparse
+import sys
+import unittest
+from unittest import TestSuite
+import requests
+
+from tests.smoke_test_case import SmokeTestCase
+from tests.unauthenticated.status_tests import StatusTests
+from tests.authenticated.azure.managed_apps_tests import ManagedAppsTests
+from tests.authenticated.billing_profiles import BillingProfileTests
+
+DESCRIPTION = """
+BPM Smoke Test
+Enter the host (domain and optional port) of the BPM instance you want to to test. This test will ensure that the BPM 
+instance running on that host is minimally functional.
+"""
+
+
+def gather_tests(is_authenticated: bool = False, azure_subscription: bool = False) -> TestSuite:
+    suite = unittest.TestSuite()
+    status_tests = unittest.defaultTestLoader.loadTestsFromTestCase(StatusTests)
+    suite.addTests(status_tests)
+    if is_authenticated:
+        profile_tests = unittest.defaultTestLoader.loadTestsFromTestCase(BillingProfileTests)
+        suite.addTests(profile_tests)
+        if azure_subscription:
+            managed_apps_tests = unittest.defaultTestLoader.loadTestsFromTestCase(ManagedAppsTests)
+            suite.addTests(managed_apps_tests)
+    return suite
+
+
+def main(main_args):
+    if main_args.user_token:
+        verify_user_token(main_args.user_token)
+
+    SmokeTestCase.BPM_HOST = main_args.bpm_host
+    SmokeTestCase.USER_TOKEN = main_args.user_token
+    ManagedAppsTests.AZURE_SUBSCRIPTION_ID = main_args.azure_sub_id
+
+    test_suite = gather_tests(main_args.user_token, main_args.azure_sub_id)
```

Comment:
As mentioned above, this would make more sense to me if you made the arguments passed booleans, like suggested below. Or perhaps return a boolean from `verify_user_token` to represent that the use has authenticated and use that for the first argument?

```
test_suite = gather_tests(main_args.user_token != null, main_args.azure_sub_id != null)